sacred/utils.py

```diff
@@ -315,6 +316,58 @@ def __init__(
         super().__init__(message, print_traceback, filter_traceback, print_usage)
 
 
+class FilteredTracebackException(tb.TracebackException):
+    def __init__(
+        self,
+        exc_type,
+        exc_value,
+        exc_traceback,
+        *,
+        limit=None,
+        lookup_lines=True,
+        capture_locals=False,
+        _seen=None,
+    ):
+        exc_traceback = self._filter_tb(exc_traceback)
+        if exc_value.__cause__:
+            exc_value.__cause__.__traceback__ = self._filter_tb(
+                exc_value.__cause__.__traceback__
+            )
+        if exc_value.__context__:
+            exc_value.__context__.__traceback__ = self._filter_tb(
+                exc_value.__context__.__traceback__
+            )
```

Comment:
I looked again at the `TracebackException` code. It looks like the causes can be chained to any depth, so it is not enough to just overwrite `__cause__.__traceback__`, we have to overwrite  `__cause__.__cause__.__traceback__`, `__cause__.__cause__.__cause__.__traceback__`, .... I start to believe that it would be easier to copy `__init__` and replace instances of `TracebackException` with `FilteredTracebackException`.

Reply:
Same for `__context__`